services/libs/integrations/src/integrations/github/processStream.ts

```diff
@@ -0,0 +1,844 @@
+      repo: data.repo,
+    })
+
+    // going through replies
+    for (const reply of record.replies.nodes) {
+      if (!('author' in reply) || !reply?.author || !reply?.author?.login) {
+        // eslint-disable-next-line no-continue
+        continue
+      }
+      const member = await prepareMember(reply.author, ctx)
+
+      // publish data
+      await ctx.publishData<GithubApiData>({
+        type: GithubActivityType.DISCUSSION_COMMENT,
+        subType: GithubActivitySubType.DISCUSSION_COMMENT_REPLY,
+        data: reply,
+        member,
+        sourceParentId: commentId,
+        repo: data.repo,
+      })
+    }
+  }
+}
+
+const handler: ProcessStreamHandler = async (ctx) => {
+  await timeout(1000)
+  const streamIdentifier = ctx.stream.identifier
+  if (streamIdentifier.startsWith(GithubStreamType.ROOT)) {
+    await processRootStream(ctx)
+  } else {
+    const data = ctx.stream.data as GithubBasicStream
+    const repo = getRepoByName(data.repo.name, ctx)
+
+    if (repo === null) {
+      await ctx.abortWithError(
+        `Stream ${ctx.stream.identifier} can't be processed since repo ${data.repo.name} is not found!`,
+        {
+          repoName: data.repo.name,
+          streamIdentifier: ctx.stream.identifier,
+        },
+      )
+    }
+
+    if (!repo.available) {
+      await ctx.abortWithError(
+        `Stream ${ctx.stream.identifier} can't be processed since repo ${data.repo.name} is not available!`,
+        {
+          repoName: data.repo.name,
+          streamIdentifier: ctx.stream.identifier,
+        },
+      )
+    }
+
+    // if all checks are passed, we can start processing
+
+    if (streamIdentifier.startsWith(GithubStreamType.STARGAZERS)) {
+      await processStargazersStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.FORKS)) {
+      await processForksStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.PULLS)) {
+      await processPullsStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.PULL_COMMENTS)) {
+      await processPullCommentsStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.PULL_REVIEW_THREADS)) {
+      await processPullReviewThreadsStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.PULL_REVIEW_THREAD_COMMENTS)) {
+      await processPullReviewThreadCommentsStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.PULL_COMMITS)) {
+      await processPullCommitsStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.ISSUES)) {
+      await processIssuesStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.ISSUE_COMMENTS)) {
+      await processIssueCommentsStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.DISCUSSIONS)) {
+      await processDiscussionsStream(ctx)
+    } else if (streamIdentifier.startsWith(GithubStreamType.DISCUSSION_COMMENTS)) {
+      await processDiscussionCommentsStream(ctx)
```

Comment:
NIT: it'd be cleaner if we split the stream identifier and do a switch-case here, what do u think?

Reply:
Yeah would be cleaner, but it's hard to ensure that all stream identifiers have the same structure, so not sure I can always make a split. That's why ifs feel safer

Reply:
Ok I checked, should be fine

Updated